### ISSUE TITLE
Unable to set up SQL

### ISSUE THREAD

**Issue by drdobler** [mRemoteNG/mRemoteNG#2425]
## Current Behavior

I've created a MSSQL DB and want to connect to it. Connection test says: OK
After applying settings it throws an unhandled exception. Unvalid objectname tblRoot
I tried the latest stable an nb build. Same error.

## Your Environment

* Version used: 1.77.1.27654,  1.76.20
* Operating System and version (e.g. Windows 10 1709 x64): Windows 10 22H2, MSSQL2014 on Server 2012R2

**Comment by drdobler** [comment]
<details><summary>Stack trace</summary>
<p>

   bei System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   bei System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   bei System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   bei System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   bei System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   bei System.Data.SqlClient.SqlDataReader.get_MetaData()
   bei System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   bei System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   bei System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   bei System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   bei System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   bei System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   bei System.Data.Common.DbCommand.ExecuteReader()
   bei mRemoteNG.Config.Serializers.MsSql.SqlDatabaseMetaDataRetriever.GetDatabaseMetaData(IDatabaseConnector databaseConnector) in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\Config\Serializers\ConnectionSerializers\MsSql\SqlDatabaseMetaDataRetriever.cs:Zeile 45.
   bei mRemoteNG.Config.Connections.SqlConnectionsLoader.Load() in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\Config\Connections\SqlConnectionsLoader.cs:Zeile 47.
   bei mRemoteNG.Connection.ConnectionsService.LoadConnections(Boolean useDatabase, Boolean import, String connectionFileName) in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\Connection\ConnectionsService.cs:Zeile 126.
   bei mRemoteNG.UI.Forms.OptionsPages.SqlServerPage.ReinitializeSqlUpdater() in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\UI\Forms\OptionsPages\SqlServerPage.cs:Zeile 82.
   bei mRemoteNG.UI.Forms.OptionsPages.SqlServerPage.SaveSettings() in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\UI\Forms\OptionsPages\SqlServerPage.cs:Zeile 72.
   bei mRemoteNG.UI.Forms.FrmOptions.btnOK_Click(Object sender, EventArgs e) in C:\jenkins\workspace\Publish\Build mRemoteNG for Release\mRemoteV1\UI\Forms\FrmOptions.cs:Zeile 128.
   bei System.Windows.Forms.Control.OnClick(EventArgs e)
   bei System.Windows.Forms.Button.OnClick(EventArgs e)
   bei System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   bei System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   bei System.Windows.Forms.Control.WndProc(Message& m)
   bei System.Windows.Forms.ButtonBase.WndProc(Message& m)
   bei System.Windows.Forms.Button.WndProc(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   bei System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

</p>
</details>

**Comment by hispeed88** [comment]
I can confirm that SQL is working: V.1.77.2 with SQL 2019. I use that for months now without any problem.

**Comment by drdobler** [comment]
Is there anything i need to know setting it up?
Docu says i just have to create a blank DB...

**Comment by hispeed88** [comment]
Yes you have create a blank DB and then you have to use a script for creating the tables and rows. 

I see this has been **removed** from the docs!

@Kvarkas 
Why has the script been removed from the docs? I

**Comment by drdobler** [comment]
I found the script via waybackmachine from a snapshot of the website from july 2022.
I can at least connect now to the DB via mRemote.
But still can't add any entries:
Couldn't save connections file as ""!
The given ColumnMapping does not match up with any column in the source or destination.

https://web.archive.org/web/20220709193121/https://mremoteng.readthedocs.io/en/latest/sql_configuration.html#mssql

**Comment by Kvarkas** [comment]
> Yes you have create a blank DB and then you have to use a script for creating the tables and rows.
> 
> I see this has been **removed** from the docs!
> 
> @Kvarkas Why has the script been removed from the docs? I

not sure, perhaps acsidentialy, will check

**Comment by eponymic** [comment]
I can't seem to find a resolution to this problem.  I have now used the above link to create tables, so thank you to @drdobler, but it still doesn't work on MySQL for me.  From the same PC running mRemoteNG I have been able to access the remote database using DataGrip and command line mysql clients.  One test used and existing installation of mysql  Ver 15.1 Distrib 5.5.50-MariaDB, for Linux (x86_64) using readline 5.1.  Another test used mysql  Ver 9.2.0 for Linux on x86_64 (MySQL Community Server - GPL).  Does anyone have any further insights into this problem?

**Comment by Kvarkas** [comment]
hi, working on that topic already.

**Comment by robertpopa22** [comment]
**Update from fork `robertpopa22/mRemoteNG`**

SQL setup has been fixed. `InitializeDatabaseSchema` + `UpgradeMssqlSchema`/`UpgradeMysqlSchema` now handle both fresh setup and upgrades automatically.

Available in **[CENSORED!] Links to external GitHub repositories are not allowed** and included in **[CENSORED!] Links to external GitHub repositories are not allowed**.
